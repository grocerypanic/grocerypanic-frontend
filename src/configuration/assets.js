const Assets = {
  favicon: `${process.env.PUBLIC_URL}/favicon.ico`,
  nonBreakingSpace: "\u00A0",
  cart: `${process.env.PUBLIC_URL}/images/cart_desktop.jpg`,
  bill: `${process.env.PUBLIC_URL}/images/bill_desktop.jpg`,
  cart_mobile: `${process.env.PUBLIC_URL}/images/cart_mobile.jpg`,
  bill_mobile: `${process.env.PUBLIC_URL}/images/bill_mobile.jpg`,
};

export default Assets;
