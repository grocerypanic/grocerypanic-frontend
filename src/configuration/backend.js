export const SafeMethods = ["GET", "HEAD", "OPTIONS", "TRACE"];
export const ItemFilters = ["preferred_stores", "shelf"];
export const FilterTag = "filter_tag";

export const Constants = {
  csrfLocalStorage: "csrf-token",
  csrfHeaderName: "X-CSRFToken",
  genericAPIError: "API Error",
  csrfErrorMessage: "Refresh csrf and try again.",
  alreadyRegistered: "User is already registered with this e-mail address.",
  minimumQuanity: 0,
  maximumQuantity: 10000,
  minimumShelfLife: 1,
  maximumShelfLife: 365 * 3,
  defaultShelfLife: 7,
  minimumPrice: 0,
  maximumPrice: 9999,
  maximumTransactionHistory: 14,
  AnalyticsCookieName: "PanicDataConsent-" + process.env.NODE_ENV.toUpperCase(),
  duplicateObjectApiError: { non_field_errors: ["Object already exists."] },
  pageLookupParam: "page",
  pageOverrideParam: "all_results",
};

export const Providers = {
  google: "google",
  facebook: "facebook",
};

export const Paths = {
  googleLogin: "/api/v1/auth/social/google/",
  facebookLogin: "/api/v1/auth/social/facebook/",
  manageShelves: "/api/v1/shelves/",
  manageStores: "/api/v1/stores/",
  manageItems: "/api/v1/items/",
  refreshCSRF: "/api/v1/auth/csrf/",
  manageTransactions: "/api/v1/transactions/",
};

const month = 30;
const year = 365;

export const ShelfLifeConstants = [
  { name: "3 Days", id: 3 },
  { name: "1 Week", id: 7 },
  { name: "2 Weeks", id: 14 },
  { name: "3 Weeks", id: 21 },
  { name: "1 Month", id: month },
  { name: "2 Months", id: month * 2 },
  { name: "3 Months", id: month * 3 },
  { name: "6 Months", id: month * 6 },
  { name: "1 Year", id: year },
  { name: "2 Years", id: year * 2 },
  { name: "3 Years", id: year * 3 },
];
