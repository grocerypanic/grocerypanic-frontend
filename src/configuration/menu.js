import Routes from "./routes";

const Options = [
  { name: "All Items", location: Routes.items },
  { name: "Shelves", location: Routes.shelves },
  { name: "Stores", location: Routes.stores },
];

export default Options;
