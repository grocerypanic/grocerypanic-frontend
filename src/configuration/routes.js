const Routes = {
  root: "/",
  menu: "/menu",
  about: "/about",
  privacy: "/privacy",
  signin: "/signin",
  shelves: "/shelves",
  splash: "/",
  stores: "/stores",
  items: "/items",
  create: "/create",
  details: "/details/:id",
  goBack: "back",
};

export const External = {
  feedback: "https://www.sharedvisionsolutions.com/#contact",
  credit: "https://www.sharedvisionsolutions.com/",
};

export default Routes;
