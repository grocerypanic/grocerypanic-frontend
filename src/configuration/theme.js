export const primary = "#006600";
export const white = "#FFFFFF";
export const secondary = "#868e96";
export const tertiary = "#808080";
export const error = "#FF0000";
export const regular = "#000000";
export const success = "#5cb85c";
export const highlight = "blue";
export const selected = "red";
export const danger = "#d9534f";
export const dark = "#495057";
export const black = "#000000";
export const warning = "#ffc107";

export const ui = {
  alertTimeout: 2000,
  resizeTimeout: 1,
};

export const navbarSize = "50px";
export const tabSize = "26px";
export const itemAttributes = {
  height: 55,
  padding: 5,
  border: 10,
  spacing: 5,
  innerHeight: 40,
  placeHolderPadding: 12,
};

export const headerMobileThreshold = 480;
export const mobileThreshold = 480;

export const graph = {
  quantityLine: "rgb(2, 117, 216)",
  changeLine: "rgb(92, 184, 92)",
  topPadding: -12,
  holdingPatternHeight: "134px",
  chartContainerHeight: "150px",
  holdingPatternColor: "secondary",
  holdingPatternScale: 1,
  holdingPatternDivHeight: 20,
  holdingPatternAnimation: "grow",
};
