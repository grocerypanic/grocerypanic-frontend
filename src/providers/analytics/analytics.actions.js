export const AnalyticsActions = {
  TestAction: { category: "Test", action: "Submitted a test action" },
  LoginError: { category: "Error", action: "Unable to login with backend" },
};
