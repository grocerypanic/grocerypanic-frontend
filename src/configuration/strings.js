import assets from "./assets";

export const Strings = {
  en: {
    SignIn: {
      Title: "Sign In",
      LoginMessageGoogle: "Login With Google",
      LoginMessageFacebook: "Login With Facebook",
      ErrorLoginFailure: "Unable to authenticate!\nPlease try again.",
      ErrorAuthExpired: "Authentication Expired!\nPlease login again.",
      ErrorDuplicateAccount:
        "You have already signed up!\nTry logging in with another social account.",
      PendingSocialConnection: "Connecting . . .",
    },
    ErrorDialogue: {
      ErrorDialogueConfirm: "OK",
      ErrorDialogueTitle: "Error!",
    },
    Copyight: {
      CopyrightDeclaration: "Copyright Â© ",
      CopyrightMessage: "Your Website",
    },
    App: {
      Suspense: "Suspense filled message...",
    },
    CookiePolicy: {
      CookieMessage:
        "This website would like to use cookies to measure how you as an individual interact with it.  We'll use this information to improve the service.",
      CookieAcceptText: "I'm ok with that",
      CookieDeclineText: "I Decline",
    },
    MainMenu: {
      Title: "Main Menu",
      HeaderTitle: "Panic: Main Menu",
      HelpText: "Make a selection to proceed.",
    },
    ItemDetails: {
      Title: "Edit Item Details",
      HeaderTitle: "Details",
      HelpText:
        "Fill out the values, click the save button to finalize.\nDon't modify quantity here.",
      NameLabel: "Name",
      QuantityLabel: "Qty",
      PriceLabel: "Price",
      StoresLabel: "Stores",
      SaveButton: `${assets.nonBreakingSpace}Save${assets.nonBreakingSpace}`,
      DeleteButton: "Delete",
      MultiSelectHelp: "",
      ShelvesDetail: "Where do you keep this at home?",
      ShelfLifeDetail: "How long does this item typically keep?",
      PerferredLocationDetails: "Where do you prefer to buy this?",
      ErrorUnselectedStore: "Choose a preferred stored.",
      SaveAction: "Saved",
      DeleteAction: "Deleting ... :(",
      Tabs: {
        Edit: "Edit",
        Stats: "Stats",
      },
      ApiError:
        "Unable to retrieve data from the API!\nDoes this content really exist?\nRetry, or change your query.",
    },
    ItemList: {
      ApiError:
        "Unable to retrieve data from the API!\nDoes this content really exist?\nRetry, or change your query.",
    },
    InventoryPage: {
      Title: "Inventory (All)",
      HeaderTitle: "Inventory",
      HelpText:
        "Click and hold the item's name for details.\nUse the create button (upper right) to add new items.\nIncrese or decrease quantity with the buttons.",
      Save: "save",
      Delete: "delete",
      PlaceHolderMessage: "You have no items yet.",
      ErrorInsufficientInventory: "You don't have that many.",
      Quantity: {
        Title: "Quantity",
        Message: "This is how many items you have in stock.",
      },
      Expired: {
        Title: "Expired",
        Message:
          "This is how many items you have in stock that may be expired.",
      },
    },
    SimpleList: {
      ValidationAlreadyExists: "This already exists",
      ValidationFailure: "Enter a valid name",
      SaveButton: "save",
      DeleteButton: "delete",
      CreatedAction: "Created",
      DeletedAction: "Deleted",
    },
    ShelfPage: {
      Title: "Your Shelves",
      HeaderTitle: "Shelves",
      PlaceHolderMessage: "You Have No Shelves Yet.",
      HelpText:
        "Click a shelf to select, click and hold to delete.\nUse the create button (upper right) to add new shelves.",
    },
    StorePage: {
      Title: "Your Stores",
      HeaderTitle: "Stores",
      PlaceHolderMessage: "You Have No Stores Yet.",
      HelpText:
        "Click a store to select, click and hold to delete.\nUse the create button (upper right) to add new stores.",
    },
    Testing: {
      GenericTranslationTestString: "GenericTranslationTestString",
      GenericMultiLineTranslationTestString:
        "GenericTranslationTestString\nGenericTranslationTestString",
    },
    PlaceHolder: {
      PlaceHolderMessage: "I'm only a placeholder.",
    },
  },
};

// Export english as default for matching during tests
export default Strings.en;
