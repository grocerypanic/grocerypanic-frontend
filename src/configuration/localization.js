import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { Strings } from "./strings";

const i18nConfig = {
  fallbackLng: {
    default: ["en"],
  },
  debug: true,
  ns: ["translation"],
  defaultNS: "translation",
  keySeparator: ".",
  interpolation: {
    escapeValue: false,
    formatSeparator: ",",
  },
  react: {
    wait: true,
  },
};

i18n.use(LanguageDetector).init({
  resources: {
    ...Strings,
  },
  ...i18nConfig,
});

export default i18n;
