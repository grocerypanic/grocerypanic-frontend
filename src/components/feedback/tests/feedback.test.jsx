import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";

import FeedBack from "../feedback.component";

import { AnalyticsActions } from "../../../providers/analytics/analytics.actions";
import { AnalyticsContext } from "../../../providers/analytics/analytics.provider";

import Strings from "../../../configuration/strings";
import { External } from "../../../configuration/routes";

const mockAnalyticsContext = {
  initialized: true,
  event: jest.fn(),
  setup: true,
};

describe("Setup Environment", () => {
  let utils;

  beforeEach(() => {
    utils = render(
      <AnalyticsContext.Provider value={mockAnalyticsContext}>
        <FeedBack />
      </AnalyticsContext.Provider>
    );
  });

  afterEach(cleanup);

  it("should render with the correct message content", () => {
    expect(utils.getByTestId("FeedBack")).toBeTruthy();
    expect(utils.getByText(Strings.FeedBack.Request1)).toBeTruthy();
  });
  it("should render with the correct external link", () => {
    const node = utils.getByTestId("FeedBackLink");
    const [text2, link, text3] = Array.from(node.childNodes);
    expect(text2.textContent).toBe(Strings.FeedBack.Request2);
    expect(link.getAttribute("href")).toEqual(External.feedback);
    expect(link.textContent).toBe(`${Strings.FeedBack.Link}`);
    expect(text3.textContent).toBe(Strings.FeedBack.Request3);
  });

  it("should handle a click on the link correctly", (done) => {
    const node = utils.getByTestId("FeedBackLink");
    const [, link] = Array.from(node.childNodes);
    fireEvent.click(link, "click");
    expect(mockAnalyticsContext.event).toBeCalledWith(
      AnalyticsActions.FeedBackLink
    );
    done();
  });

  it("should match the snapshot on file (styles)", () => {
    expect(utils.container.firstChild).toMatchSnapshot();
  });
});
