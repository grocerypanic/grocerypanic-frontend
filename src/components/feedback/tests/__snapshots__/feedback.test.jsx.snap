// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment should match the snapshot on file (styles) 1`] = `
.c0 {
  padding-top: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 0.7em;
  color: #808080;
}

.c1 {
  cursor: pointer;
  color: #006600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  class="c0"
  data-testid="FeedBack"
>
  <span>
    I would appreciate any feedback or suggestions!
  </span>
  <span
    data-testid="FeedBackLink"
  >
    Please leave themÂ 
    <a
      class="c1"
      href="https://www.sharedvisionsolutions.com/#contact"
      target="_blank"
    >
      here
    </a>
    .
  </span>
</div>
`;
